### ISSUE TITLE
Stream stops after a few minutes

### ISSUE THREAD

**Issue by yagogr98** [blaise-io/acelink#79]
Hi! first of all thankyou for this beautiful tool we have!

I'm having a problem with a fresh install of Acelink in mac OS 14.3.1.
All streams stop after a few minutes of playback. This happens on a MacBook Pro M1 and shows this error on the docker log.

2024-02-22 09:00:17 2024-02-22 08:00:17,352|BGTaskQueueThread-1|acestream.notifications|cannot build from server options: options={}

Thanks in advance :)

**Comment by alvertbalero** [comment]
Same here! Stream stop after about 10-15 min. every time. Mac mini M1. Mac OS 14.1.1
Thanks!

**Comment by AirmakZ** [comment]
Exactly same here.
MacBook Air M1 
macOS 14.3.1

**Comment by antoniiolpzzz** [comment]
Same here as well, I think the problem is were all from Spain, am I wrong?

Supposedly new acestream versions fix that, that would be good to get a new ace link version with the latest acestream, maybe it would fix it.

**Comment by dani2992** [comment]
It is a problem related to the banning of operators in Spain to acestream. This [thread](https://forum.acestream.media/t/acestream-disconnects-every-12-15-minutes/3766/28) talks about it and at the end it seems to be solved in acestream version 3.2. I don't know if this modification can also be incorporated in acelink?

**Comment by AirmakZ** [comment]
I was investigating a little about it.
Seems that basically owner shall update the docker install downloaded by AceStream. It shall be done in the Docker images server. Not the app itself. It's good news. They don't need even to release a new version.

Also I tried to compile the application from my side trying to change the Docker image Acelink is pointing but at least in a first attempt it's not working.
Anyway I could not find an updated v3.2 AceStream Engine so even I make it works we will face same issue.

**Comment by blaise-io** [comment]
From the comments in this thread I have no idea what the actual issue is. But you could re-build the Docker image locally and see if that fixes the issue. Dockerfile is in this repo.

**Comment by blaise-io** [comment]
Thanks, I'll check whether I can get 3.2 working in Docker.

**Comment by blaise-io** [comment]
Last Linux version supported by Acestream is 3.1.75rc4. @AirmakZ where did you read 3.2 may fix the issue?

**Comment by AirmakZ** [comment]
I could read about in a quite big telegram group where people is updating to 3.2. Surely they are using Windows version.
Also @dani2992 commented about this thread (https://forum.acestream.media/t/acestream-disconnects-every-12-15-minutes/3766/28).
I don't know about different platforms releases timeline I supposed they are parallel.
If 3.2 is not yet released sure nothing we can do other than wait and appreciate your time and dedication. Of course.

**Comment by blaise-io** [comment]
Can you try this build? [Ace.Link.2.0.7-acestream-3.1.75rc4.dmg.zip](https://github.com/blaise-io/acelink/files/14473604/Ace.Link.2.0.7-acestream-3.1.75rc4.dmg.zip) it's the best I can do for Ace Link, otherwise you could try asking the Acestream team. Seems that 3.1.74 still has the issue (https://forum.acestream.media/t/acestream-disconnects-every-12-15-minutes/3766/10) so I don't have high hopes unfortunately.

**Comment by AirmakZ** [comment]
Hi @blaise-io, playing now. Let's see.
Yeah maybe pushing them to release 3.2 branch for Linux.

**Comment by AirmakZ** [comment]
@blaise-io, as expected. 3.1.75rc4 after 12 minutes it stops.

2024-03-03 17:17:29 2024-03-03 16:17:29,713|PlaylistTaskManagerThread-32|acestream.TaskManager|started

2024-03-03 17:17:46 2024-03-03 16:17:46,039|ServiceTaskQueueThread-2|acestream.Session|sync_time: no response

2024-03-03 17:29:35 2024-03-03 16:29:35,910|BGTaskQueueThread-1|acestream.notifications|cannot build from server options: options={}

**Comment by AirmakZ** [comment]
They mention in the forum "Version 3.2 for Linux will be released soon, sorry for the delay."
So maybe it will be available during next days.

**Comment by blaise-io** [comment]
They also said a macOS version would be released 10 yrs ago 😅

**Comment by raulfdeztdo** [comment]
We still have the same problem, after 10-15 mins it crashes and we have to restart the docker container. Is there any news to solve the bug?

**Comment by AlFontal** [comment]
The 10-15 min "crash" seems to be a known issue and happens due to Spanish ISPs blocking Acestream's ad servers which leads to Acestream engine shutting down after a while of not being able to connect to them. I paid the "premium" acestream and it solves it when running it in my android TV or android phone, but I am unable to do so through acelink.

Can I access the acestream engine UI somehow while running it through docker via acelink to access my premium account anyway? I think this problem will not be solvable otherwise until further updates in the acestream engine overcome the blocking by Spanish ISPs (if that ever happens).

**Comment by Ki1L3r** [comment]
@blaise-io seems like they have already released the 3.2 version for Linux https://docs.acestream.net/products/

**Comment by AirmakZ** [comment]
Yes. You’re right @Ki1L3r. Version 3.2.3 is available.
@blaise-io, if there is a way to build a test version as we done in the past we can confirm quite fast if it works.
Thanks!

**Comment by zer00zz** [comment]
Did a quick build using acestream 3.2.3. I'm checking It right now, no stopping so far. I've created a pull request with the changes here https://github.com/blaise-io/acelink/pull/87.

Quick build if you want to test it: [acelink+acestream3.2.3](https://github.com/zer00zz/acelink/releases/tag/2.0.6-acestream-3.2)

Will remove this build once an official release is ready.

A disfrutar!

**Comment by blaise-io** [comment]
I was on holiday, sorry for my late reply. Can you verify whether this works @zer00zz @amanzan? [Ace.Link.2.1.0-beta1.dmg.zip](https://github.com/blaise-io/acelink/files/15220599/Ace.Link.2.1.0-beta1.dmg.zip). It's using a new [blaiseio/acelink:2.1.0](https://hub.docker.com/layers/blaiseio/acelink/2.1.0/images/sha256-b174dd6e6899e881aeec2dc7705a71e8e56dd57bf830e88b0d163e6ada304452) docker image. If it works, I'll add it as a new release.

**Comment by zer00zz** [comment]
@blaise-io thanks for the time and sorry for all the fuzz! I'll check that one tomorrow.

**Comment by amanzan** [comment]
> I was on holiday, sorry for my late reply. Can you verify whether this works @zer00zz @amanzan? [Ace.Link.2.1.0-beta1.dmg.zip](https://github.com/blaise-io/acelink/files/15220599/Ace.Link.2.1.0-beta1.dmg.zip). It's using a new [blaiseio/acelink:2.1.0](https://hub.docker.com/layers/blaiseio/acelink/2.1.0/images/sha256-b174dd6e6899e881aeec2dc7705a71e8e56dd57bf830e88b0d163e6ada304452) docker image. If it works, I'll add it as a new release.

@blaise-io, seems to work fine! Thanks!

**Comment by janderklanderr** [comment]
It doesn´t work for me

**Comment by blaise-io** [comment]
@janderklanderr please file a new issue with [logs](https://github.com/blaise-io/acelink?tab=readme-ov-file#view-ace-link-logs)?

**Comment by zer00zz** [comment]
@blaise-io I’ve used it without issue. I think you can close this issue. Thanks again for this project 🚀
